src/spikeinterface/core/basesorting.py

```diff
@@ -269,35 +269,40 @@ def get_total_num_spikes(self):
         )
         return self.count_num_spikes_per_unit()
 
-    def count_num_spikes_per_unit(self) -> dict:
+    def count_num_spikes_per_unit(self, output="dict"):
```

Comment:
I think it would be better to just make another function for this behavior.

What is the advantage of doing it with a keyword?
The disavatnages are:
* Harder to discover
* More complicated output type which makes harder to reason about code.
* Adds complexity to the function.

Functions should do one thing and do it well. Specially at the core. Or make two private methods that do the specific task and make this a dispatcher with options.

Reply:
For me the main advantage is internal use of `sorting.count_num_spikes_per_unit(output="array")`
replacing at several places the uggly  `np.array(list(sorting.count_num_spikes_per_unit().values()))`

Reply:
But you can just have another method that will allow you to do exactly the same:

`sorting.count_num_spikes_per_unit_array' instead of `np.array(list(sorting.count_num_spikes_per_unit().values()))`

I think the method is a good idea.
I think mixing this new logic in the old method is not for the reasons stated above.

 I am asking: why do you want to mix the logic to produce counts as arrays and counts as dict in the same fuction? What's the advantage of it? Is than an aesthethical thing? You dislike having more functions? You personally find keywords easier to discover than functions? I am looking for your reasoning here.

"Functions should do only one thing and do it well" seems like a really good principle to me that is more or less widely accepted. Design rules can -and sometimes should- be broken, but I think we should have a good reason to do it. Which one is here?